File: nextjs_react_lotto/app/lib/db/lotto_result.ts
'use server'

import { LottoDrawResult } from "../domain";
import { sqliteDB } from "./sqlite";

export async function getLottoResult(drwNo: number) : Promise<LottoDrawResult | undefined> {
    return new Promise( (resolve, reject) => {
        sqliteDB.get('SELECT * FROM lotto_result WHERE drw_no=?',[drwNo],(err,row) =>{
            if(err){
                return reject(err)
            }
            if( row ) {
                resolve({
                    drwNo : row['drw_no'],
                    drwNoDate: row['drwno_date'],
                    totSellamnt: row['tot_sell_amnt'],
                    firstWinamnt: row['first_win_amnt'],
                    firstPrzwnerCo: row['first_prz_wner_co'],
                    firstAccumamnt: row['first_win_tot_amnt'],
                    drwtNo1: row['drwt_no1'],
                    drwtNo2: row['drwt_no2'],
                    drwtNo3: row['drwt_no3'],
                    drwtNo4: row['drwt_no4'],
                    drwtNo5: row['drwt_no5'],
                    drwtNo6: row['drwt_no6'],
                    bnusNo: row['bnusNo'],
                } as LottoDrawResult)
            } 
            resolve(null)
        })
    })
}

const lotto_result_insert_sql = `
INSERT INTO lotto_result (
    drw_no,
    drwt_no1,
    drwt_no2,
    drwt_no3,
    drwt_no4,
    drwt_no5,
    drwt_no6,
    bnusNo,
    drwno_date,
    tot_sell_amnt,
    first_win_amnt,
    first_prz_wner_co,
    first_win_tot_amnt
) values(?,?,?,?,?,?,?,?,?,?,?,?,?)
`
export async function insertLottoResult(lottoResult : LottoDrawResult){
    return new Promise( (resolve, reject) => {
        sqliteDB.run(lotto_result_insert_sql,[
            lottoResult.drwNo,
            lottoResult.drwtNo1,
            lottoResult.drwtNo2,
            lottoResult.drwtNo3,
            lottoResult.drwtNo4,
            lottoResult.drwtNo5,
            lottoResult.drwtNo6,
            lottoResult.bnusNo,
            lottoResult.drwNoDate,
            lottoResult.totSellamnt,
            lottoResult.firstWinamnt,
            lottoResult.firstPrzwnerCo,
            lottoResult.firstAccumamnt,
        ],(err) => {
            if(err){
                reject(err)
            }
            resolve(null)
        })
    })
}

export async function deleteLottoResult(drwNo: number){
    return new Promise( (resolve, reject) => {
        sqliteDB.run('DELETE FROM lotto_result WHERE drw_no=?',[drwNo], (err) => {
            if(err){
                reject(err)
            }
            resolve(null)
        })
    })
}

File: go_lottoresult/app/data/models/lotto-result.go
package models

import (
	"encoding/json"
)

type LottoDrawResult struct {
	DrwNo          int    `gorm:"primaryKey;column:drw_no" json:"drwNo"`
	DrwNoDate      string `gorm:"column:drw_no_date" json:"drwNoDate"`
	DrwtNo1        int    `gorm:"column:drwt_no1" json:"drwtNo1"`
	DrwtNo2        int    `gorm:"column:drwt_no2" json:"drwtNo2"`
	DrwtNo3        int    `gorm:"column:drwt_no3" json:"drwtNo3"`
	DrwtNo4        int    `gorm:"column:drwt_no4" json:"drwtNo4"`
	DrwtNo5        int    `gorm:"column:drwt_no5" json:"drwtNo5"`
	DrwtNo6        int    `gorm:"column:drwt_no6" json:"drwtNo6"`
	BnusNo         int    `gorm:"column:bnus_no" json:"bnusNo"`
	TotSellAmnt    uint64 `gorm:"column:tot_sell_amnt" json:"totSellamnt"`
	FirstWinAmnt   uint64 `gorm:"column:first_win_amnt" json:"firstWinamnt"`
	FirstPrzWnerCo int    `gorm:"column:first_prz_wner_co" json:"firstPrzwnerCo"`
	FirstAccumAmnt uint64 `gorm:"column:first_accum_amnt" json:"firstAccumamnt"`
	ReturnValue    string `gorm:"column:return_value" json:"returnValue"`
}

func (l LottoDrawResult) TableName() string {
	return "lotto_draw_result"
}

func (l LottoDrawResult) ToJson() string {
	result, err := json.Marshal(l)
	if err != nil {
		return ""
	}
	return string(result)
}

func GetLottoDrawResultFronJsonString(jsonStr string) (LottoDrawResult, error) {
	var lottoDrawResult LottoDrawResult
	err := json.Unmarshal([]byte(jsonStr), &lottoDrawResult)
	return lottoDrawResult, err
}
